staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -113,17 +120,36 @@ type RoundTripperConfig struct {
 	// PingPeriod is a period for sending SPDY Pings on the connection.
 	// Optional.
 	PingPeriod time.Duration
+	// UpgradeTransport is a subtitute transport used for dialing. If set,
+	// this field will be used instead of "TLS" and "Proxier" for connection creation.
+	// Optional.
+	UpgradeTransport http.RoundTripper
```

Comment:
in NewRoundTripperWithConfig, return an error if UpgradeTransport is non-nil and TLS or Proxier are non-nil (will require plumbing error returns back out of those constructors to call sites, but I think there's only one non-test callsite)

Reply:
Fixed.